### PR TITLE
Clarify COLOR and TEXTURE in CanvasItem shaders

### PR DESCRIPTION
**Pull request by tetrapod00**
Clarifies how to read either the main texture, the vertex `COLOR`, or both in CanvasItem shaders.

With this PR, all three options available (only `COLOR`, only `TEXTURE`, or both) are cleanly listed with snippets. Also, the role of `COLOR` in `fragment()` and `vertex()` is also clearly listed. This is slightly redundant with the tables themselves but worth the increase in clarity.

I've seen a fair bit of confusion about this in help chats, also people looking for the old `MODULATE` value from 3.x. This should help with both issues.

While I was here I corrected one use of **bold** instead of ``CODE`` styling in one of the tables.

### CODE REVIEW

File: tutorials/shaders/shader_reference/canvas_item_shader.rst [hunk 1]
```diff
@@ -99,60 +99,105 @@ is usually:
...
-by default. However, when a custom ``fragment()`` function is attached to these nodes,
-the texture lookup needs to be done manually. Godot provides the texture color
-in the ``COLOR`` built-in variable multiplied by the node's color. To read the
-texture color by itself, you can use:
+COLOR and TEXTURE
+~~~~~~~~~~~~~~~~~
+
+The built-in variable ``COLOR`` is used for a few things:
+
+  - In the ``vertex()`` function, ``COLOR`` contains the from the vertex primitive
```
Comment (mhilbrunner) [inline tutorials/shaders/shader_reference/canvas_item_shader.rst:157]: Missing word between `the from`

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you!

**Comment by tetrapod00** [comment]
This one would be *nice* to cherrypick, but as an (extensive) clarification, it's non-essential. Was the problem the table being reformatted?

**Comment by tetrapod00** [comment]
Cherrypicked to 4.3 in https://github.com/godotengine/godot-docs/pull/10356.

**Pull request merged**